python-sdk/pyproject.toml

```diff
@@ -87,7 +87,7 @@ ftp = [
     "apache-airflow-providers-ftp>=3.0.0",
     "smart-open>=5.2.1",
 ]
-openlineage = ["openlineage-airflow>=0.17.0"]
+openlineage = ["apache-airflow-providers-openlineage>=1.4.0"]
```

Comment:
Note that the OpenLineage provider only supports Airflow versions greater than or equal to 2.7.0.
For compatibility with older Airflow versions, wouldn't there be dependency conflict on strict requirement with [openlineage] extra?

Reply:
yes, thanks a lot @JDarDagran for the suggesiton

Bumping up min Airflow version to 2.7 and min Python version to 3.8

Reply:
most of the airflow provider has `apache-airflow>=2.6.0` as a dependency. hopping bumping 2.7.0  will not impact users

Reply:
We will release a minor version of Astro SDK and it should not cause a problem IMHO. If users would like to use the new minor version, they would need to have Airflow 2.7. For earlier versions, they can continue using prior versions of Astro SDK.

Similar fashion is followed in Airflow OSS providers when bumping versions for dependencies

Reply:
Besides we cannot allow usage of `apache-airflow-providers-openlineage` without bumping Airflow to 2.7